cmd/rootMain.go

```diff
@@ -117,79 +145,34 @@ Try 'help'`)
 	)
 	comm.Dir = mountpath
 
-	// Inspired by https://blog.nelhage.com/2011/02/changing-ctty/. After the child shell has
-	// started, we re-parent the Wash daemon to be in a child session of the shell so that when Wash
-	// prompts for input it's within the TTY of that terminal.
 	if startErr := comm.Start(); startErr != nil {
 		cmdutil.ErrPrintf("%v\n", startErr)
 		return exitCode{1}
 	}
 
-	codeCh := make(chan int)
-	go func() {
-		if runErr := comm.Wait(); runErr != nil {
-			if exitErr, ok := runErr.(*exec.ExitError); ok {
-				codeCh <- exitErr.ExitCode()
-			} else {
-				cmdutil.SafeErrPrintf("%v\n", runErr)
-				codeCh <- 1
-			}
-		} else {
-			codeCh <- 0
-		}
-		close(codeCh)
-	}()
+	// If interactive (when we might prompt the user for input, such as security tokens), create a
+	// new session. If not interactive, calling setsid is pointless and might fail.
+	if plugin.IsInteractive() {
+		if _, err := unix.Setsid(); err != nil {
```

Comment:
This is a lot simper.

Reply:
Only worked after I started ensuring the daemon wasn't leader of a process group. Put that note in the commit message.